src/util.cc

```diff
@@ -471,51 +479,63 @@ std::string WideToUtf8(const WStringPiece& w_string) {
   return u8_string;
 }
 
-void WideToUtf8(const WStringPiece& w_string, char* buff,
-                size_t buffer_length) {
-  if (w_string.len_ == 0)
-    return;
-
-  // Make sure that the string size can be casted to an int.
-  assert(!(w_string.size() > INT_MAX));
-
-  int size_needed = WideCharToMultiByte(CP_UTF8, 0, w_string.wstr_,
-                                        static_cast<int>(w_string.size()), NULL,
-                                        0, NULL, NULL);
-  if (size_needed == 0)
-    Fatal("Failed conversion to UTF-8 string");
-
+void ConvertWideToUtf8(const WStringPiece& w_string, int size_needed,
+                       char* buff, size_t buffer_length) {
   int u8_size_needed = WideCharToMultiByte(CP_UTF8, 0, w_string.wstr_,
                                            static_cast<int>(w_string.size()),
                                            buff, size_needed, NULL, NULL);
 
-  // WideCharToMultiByte returns zero if buff is not large enough, modified for the null termination.
-  if (u8_size_needed == 0 || (u8_size_needed > (buffer_length-1)))
+  // WideCharToMultiByte returns zero if buff is not large enough,
+  // modified for the null termination.
+  if (u8_size_needed == 0 || (u8_size_needed > (buffer_length - 1)))
     Fatal("Failed conversion to UTF-8 string");
 
   // The conversion does not include the nulltermination.
   buff[u8_size_needed] = '\0';
 }
 
+void WideToUtf8(const WStringPiece& w_string, char* buff,
+                size_t buffer_length) {
+  if (w_string.len_ == 0)
+    return;
+
+  int size_needed = GetUtf8SizeNeeded(w_string);
+  ConvertWideToUtf8(w_string, size_needed, buff, buffer_length);
+}
+
 char** convertCommandLine(int argc, wchar_t** wargv) {
-  std::wstring w_argString;
-  std::string argString;
-  char** argv = new char*[argc + 1];
+  size_t total_size = 0;
+  std::vector<int> sizes;
```

Comment:
std::vector allocates.

You shouldn't need to keep track of the sizes beyond the overall total.

Please rework.